### PR TITLE
feat(doctor): add doctor client code and doctor api call

### PR DESCRIPTION
**Pull request by smothiki**

### CODE REVIEW

File: pkg/swagger/models/version.go [hunk 1]
```diff
@@ -5,7 +5,6 @@ package models
 
 import (
 	strfmt "github.com/go-swagger/go-swagger/strfmt"
-	"github.com/go-swagger/go-swagger/swag"
```
Comment (kmala) [inline pkg/swagger/models/version.go]: why is this removed?

File: pkg/swagger/models/component_version.go [hunk 2]
```diff
@@ -1,7 +1,5 @@
 package models
 
-import "github.com/go-swagger/go-swagger/strfmt"
```
Comment (kmala) [inline pkg/swagger/models/component_version.go]: why is this removed?

File: api/swagger-spec/swagger.yml [hunk 3]
```diff
@@ -52,6 +52,26 @@ paths:
           description: unexpected error
           schema:
             $ref: "#/definitions/error"
+  /v3/doctor/{train}/{release}/{uuid}:
```
Comment (jackfrancis) [inline api/swagger-spec/swagger.yml]: @slack do we want to concern ourselves namespacing deis doctor submissions with version info (`train` and `release` here)? we can probably get away with just `/v3/doctor/{uuid}` (where `uuid` is the id of the generated deis doctor debug data)

File: pkg/swagger/models/version.go (hunk 1 above)
Comment (smothiki) [inline pkg/swagger/models/version.go]: might me a mistake from my atom editor I'll cross check again

File: api/swagger-spec/swagger.yml (hunk 3 above)
Comment (slack) [inline api/swagger-spec/swagger.yml]: `/v3/doctor/{uuid}` feels appropriate. Can't think of a reason why we'd want to scope by train and release, unless we can't make the API for the payload submission generic enough to accommodate change over time. I'd hate to see a "doctor" API that had to be revved for every release.

Comment (jackfrancis) [inline api/swagger-spec/swagger.yml]: I agree that each deis doctor "report" will be a unique record, and we only need an appropriate unique ID strategy (UUID qualifies). We can build other endpoints that match particular kinds of queries, like say, "all reports sent from a single cluster"; but for submissions, we don't need to worry about scoping them to particular versions or the originating cluster.

Comment (smothiki) [inline api/swagger-spec/swagger.yml]: we should somehow know what version doctor info the user is sending? if not here we should include that in doctorInfo struct ?

Comment (jackfrancis) [inline api/swagger-spec/swagger.yml]: We should include deis component versions (including workflow-manager) in the cluster metadata.

File: api/swagger-spec/swagger.yml [hunk 4]
```diff
@@ -275,6 +301,13 @@ definitions:
         type: array
         items:
           $ref: "#/definitions/componentVersion"
+  doctorInfo:
+    type: object
+    required:
+      - cluster
+    properties:
+      cluster:
+        $ref: "#/definitions/cluster"
```
Comment (jackfrancis) [inline api/swagger-spec/swagger.yml]: I think we should create a new model here. Maybe call it `doctorData`. It would look something like this:

```
{
    id: {uuid}
    cluster_id: {cluster id}
    components: {componentVersion}
    data: {generic object}
}
```

The `generic object` data property above would evolve over time to contain the payload of the deis doctor report; the `components` property would contain the basic deis cluster version metadata.

@slack might want to sanity check this proposal!

Comment (smothiki) [inline api/swagger-spec/swagger.yml]: we can get uuid from the url params. so avoided here . if you want me to add it here then URL has to change to just /doctor

Comment (smothiki) [inline api/swagger-spec/swagger.yml]: >{
    id: {uuid}
    cluster_id: {cluster id}
    components: {componentVersion}
    data: {generic object}
}


That's basically splitting cluster data structure

Comment (jackfrancis) [inline api/swagger-spec/swagger.yml]: That's fair to exclude the report id from the data model. I'd prefer to keep it in the route endpoint so our GET route looks the same as our POST route and we can have CRUD nirvana.

Comment (smothiki) [inline api/swagger-spec/swagger.yml]: Yes 
Although I haven;t wrote and get route but its the same and display doctor data struct. fair enough ?

Comment (jackfrancis) [inline api/swagger-spec/swagger.yml]: 👍

**Comment by jackfrancis** [comment]
closes #76
